### DISCUSSION TITLE
I am struggling to figure out how to get my mainsail OS and Raspberry Pi 3 setup to print to the Neptune 3 Pro

### DISCUSSION THREAD

**Discussion by calebthenoble** [mainsail-crew/mainsail#1845] [Help] [answered]
I am struggling to figure out how to get my mainsail OS and Raspberry Pi 3 setup to print to the Neptune 3 Pro. The issue is that it says that it is 'Unable to open file'. I am able to upload and look at the gcode in the gcode viewer. I don't understand what I am doing wrong. Any help is appreciated!

Rgds, 
CalebtheNoble

**Comment by meteyou** [comment]
Pls upload your klippy.log

**Reply by calebthenoble** [reply]
It isn't even opening now, Im gonna try and see if I can reflash it with mainsail and move from there.

**Comment by calebthenoble** [comment, marked as answer]
I reflashed MainSail and got things working, also found that the file directory I had made was incorrect. I thought I had done that right but was quite wrong and found that out once I had reflashed the pi but a lot of the other issues that started cropping up went away.

**Reply by meteyou** [reply]
thx for your feedback! Happy Printing!
